src/RIEBase.js

```diff
@@ -33,11 +34,15 @@ export default class RIEBase extends React.Component {
     };
 
     doValidations = (value) => {
+        let result;
         if(this.props.validate) {
-            this.setState({invalid: !this.props.validate(value)});
+            result = this.props.validate(value);
         } else if (this.validate) {
-            this.setState({invalid: !this.validate(value)});
+            result = this.validate(value);
         }
+        this.setState({invalid: !result});
+
+        return result;
```

Comment:
react's `setState` is async, so you can not get the new state in the following sync logics. change it to a local variable and return it directly.